### ISSUE TITLE
Signal type: int64 or uint64 or Enum?

### ISSUE THREAD

**Issue by rcoreilly** [goki/ki#20]
First, `NodeSignals` is not a bitflag -- just a regular enum.  Fixes several tests.

Currently, the `RecvFunc` specifies `sig int64`.  Should we change that to `enums.Enum` so it automagically works for any kind of enum, without having to do the type conversion that all such code is currently doing?

The cost of doing so is that an interface is always a pointer, so we're changing from a simple basic data value to a pointer everywhere.   This could put more of a drag on the GC.

I don't think that cost is worth it for a single line conversion -- it is also useful to get the specific enum instead of a generic Enum, so you have visibility in IDE about possible values etc.

So, bottom line: keep it as it is?

**Comment by rcoreilly** [comment]
See #22

**Comment by rcoreilly** [comment]
Done in #22
